test/libsolidity/semanticTests/builtins/balance_other_contract.sol

```diff
@@ -0,0 +1,23 @@
+contract Other {
+    constructor() payable {
+    }
+    function getAddress() public returns (address) {
+        return address(this);
+    }
+}
+contract ClientReceipt {
+    Other other;
+    constructor() payable {
+        other = new Other{value:500}();
+    }
+    function getAddress() public returns (address) {
+        return other.getAddress();
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor(), 1000 wei ->
+// contract.balance() -> 500
```

Comment:
I think I would prefer a syntax that is more different from regular contract function calls, so that it stands out better. We already have `storage: empty` - why not add `balance: <balance>`?

Reply:
I just used the regular function call syntax here, because it was the easiest to use. We could also think about an alternative syntax for build-ins. I was also thinking that it may become annoying that there is always the need to specify the full function signature e.g. `logs.logTopic(uint256,uint256): 0, 1 -> 0x1212121212121212121212121212120000000012`, e.g something like `logs.logTopic(0, 1) ->` would be probably easier to use.

What syntax would you prefer? The idea behind build-ins is to be able to easily plugin new "commands" into isoltest, without the need to touch any parser specific parts. 

Maybe `storage` could become a builtin too.

Reply:
Did we discuss that? I tihnk `balance: <balance>` and `balance <address>: <balance>` would be nice.

Reply:
I think we did not really discuss this. I didn't update this PR yet. However, the syntax will be defined by #10867, that means it will look like `balance -> <balance>` and `balance: <address> -> <balance>`. What are you thinking about this?

Reply:
Looks good!

Reply:
@aarlt this PR is using `contract_balance` for the first one. Is that changed from the other PR?